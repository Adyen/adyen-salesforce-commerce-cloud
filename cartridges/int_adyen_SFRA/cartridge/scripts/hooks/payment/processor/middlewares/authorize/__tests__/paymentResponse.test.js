"use strict";

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _iterableToArrayLimit(arr, i) { if (typeof Symbol === "undefined" || !(Symbol.iterator in Object(arr))) return; var _arr = []; var _n = true; var _d = false; var _e = undefined; try { for (var _i = arr[Symbol.iterator](), _s; !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

var BasketMgr = require('dw/order/BasketMgr');

var paymentResponseHandler = require('../paymentResponse');

var paymentInstrument;
var orderNumber = 'mocked_order_number';
beforeEach(function () {
  var _BasketMgr$toArray = BasketMgr.toArray();

  var _BasketMgr$toArray2 = _slicedToArray(_BasketMgr$toArray, 1);

  paymentInstrument = _BasketMgr$toArray2[0];
});
describe('Payment Response Handler', function () {
  it('should get 3ds2 response', function () {
    var result = {
      threeDS2: true,
      resultCode: 'mocked_result_code',
      fullResponse: {
        action: 'mocked_action'
      }
    };
    var response = paymentResponseHandler(paymentInstrument, result, orderNumber);
    expect(response).toMatchSnapshot();
  });
  it('should get redirect response with payment data signature', function () {
    var result = {
      threeDS2: false,
      resultCode: 'mocked_result_code',
      paymentData: 'mocked_payment_data',
      redirectObject: {
        url: 'mocked_redirect_url'
      }
    };
    var response = paymentResponseHandler(paymentInstrument, result, orderNumber);
    expect(response).toMatchSnapshot();
  });
  it('should get redirect response with md signature', function () {
    var result = {
      threeDS2: false,
      resultCode: 'mocked_result_code',
      paymentData: 'mocked_payment_data',
      redirectObject: {
        data: {
          MD: 'mocked_redirect_MD'
        },
        url: 'mocked_redirect_url'
      }
    };
    var response = paymentResponseHandler(paymentInstrument, result, orderNumber);
    expect(response).toMatchSnapshot();
  });
});