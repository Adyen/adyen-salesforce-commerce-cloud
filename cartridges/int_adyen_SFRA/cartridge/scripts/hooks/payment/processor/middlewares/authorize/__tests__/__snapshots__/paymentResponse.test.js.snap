// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Payment Response Handler should get 3ds2 response 1`] = `
Object {
  "action": "\\"mocked_action\\"",
  "resultCode": "mocked_result_code",
  "threeDS2": true,
}
`;

exports[`Payment Response Handler should get redirect response with md signature 1`] = `
Object {
  "authorized": true,
  "authorized3d": true,
  "orderNo": "mocked_order_number",
  "paymentInstrument": Object {
    "custom": Object {
      "adyenPaymentData": "mocked_payment_data",
    },
    "paymentMethod": "mocked_method",
    "paymentTransaction": Object {
      "paymentProcessor": "mocked_payment_processor",
    },
    "setCreditCardExpirationMonth": [MockFunction],
    "setCreditCardExpirationYear": [MockFunction],
    "setCreditCardNumber": [MockFunction],
    "setCreditCardToken": [MockFunction],
    "setCreditCardType": [MockFunction],
  },
  "redirectObject": Object {
    "data": Object {
      "MD": "mocked_redirect_MD",
    },
    "url": "mocked_redirect_url",
  },
  "signature": "ct_url __ ocked_redirect_MD",
}
`;

exports[`Payment Response Handler should get redirect response with payment data signature 1`] = `
Object {
  "authorized": true,
  "authorized3d": false,
  "orderNo": "mocked_order_number",
  "paymentInstrument": Object {
    "custom": Object {
      "adyenPaymentData": "mocked_payment_data",
    },
    "paymentMethod": "mocked_method",
    "paymentTransaction": Object {
      "paymentProcessor": "mocked_payment_processor",
    },
    "setCreditCardExpirationMonth": [MockFunction],
    "setCreditCardExpirationYear": [MockFunction],
    "setCreditCardNumber": [MockFunction],
    "setCreditCardToken": [MockFunction],
    "setCreditCardType": [MockFunction],
  },
  "redirectObject": Object {
    "url": "mocked_redirect_url",
  },
  "signature": "ct_url __ ocked_payment_data",
}
`;
